### ISSUE TITLE
LO extension: Deactivated Toolbar opens after restart of LO

### ISSUE THREAD

**Issue by FredKruse** [languagetool-org/languagetool#10348]
The LT toolbar opens after restarting LO even if it was previously deactivated.

**Comment by FredKruse** [comment]
@ColeBantam
The issue is solved. Please test it with the snapshot tomorrow

**Comment by ColeBantam** [comment]
👍 Toolbar stays deactivated after restart & position is saved when enabled
👎 GUI Issues remain when Toolbar is enabled
👎 disabling single Buttons in Toolbar not working (coming back after restart)

I noticed another interesting thing with the black-screen issues:
When I disable some buttons in the toolbar and restart LibO, the GUI is normal for a moment and the deactivated buttons in the Toolbar are still gone. THEN the Buttons return and the GUI goes black ...

**Comment by FredKruse** [comment]
I found a way to solve both remaining problems.
Please test it tomorrow and give me feedback.

**Comment by ColeBantam** [comment]
In 'LanguageTool 6.4-SNAPSHOT (2024-02-26 18:59:22 +0100, b0a2047)' the Toolbar looks good. No Black-GUI Issues anymore and Icons that were removed from the toolbar stay removed :)

Only remaining minor glitch is, that I still need to put the toolbar into position twice. Its not remembered after first try.

And it would be great if toolbar position isn't forgotten on every LT-Update.

**Comment by ColeBantam** [comment]
Today I noticed another issue with LT 3.4 Snapshots. Scrollbars are missing with LT 3.4 Snapshots installed. No matter if the toolbar is activated or not. When I open a new writer-document, the scrollbars are missing. Even on large documents with multiple pages, there are no scrollbars. Only after zooming in and out, the scrollbars come back. With LT 3.3.0 this issue does not exist, so another classic regression.

PS: tested with LanguageTool-20240226-snapshot and LanguageTool-20240304-snapshot.

**Comment by FredKruse** [comment]
As I've noted elsewhere, updating the LibreOffice interface is a LibreOffice problem that only occurs on Windows (by the way, if you wait long enough, the update will happen after several seconds).
I've now tried to find a workaround. In most cases, this should work. Please test it and give me feedback.

**Comment by ColeBantam** [comment]
Scrollbars are back in current snapshot :)

Only remaining "issue" is, that the toolbar is in its default location after every LT-Update.

**Comment by ColeBantam** [comment]
Update: Still on Snapshot 2024-03-11 18:59:47 +0100, 2dbbc1a. This Snapshot did seem to solve the scrollbar issue, but sadly the black-gui issue is back and also a couple of crashes:
![crash1](https://github.com/languagetool-org/languagetool/assets/34094836/571af306-ce74-4ba8-81c6-196236d27add)
![crash2](https://github.com/languagetool-org/languagetool/assets/34094836/b7126f5d-8ccf-460a-8132-5db943523de5)

**Comment by FredKruse** [comment]
Thank you for testing. I think it was a problem with a calc document, right? The problems should be solved. Please test it with the tomorrow's snapshot.

**Comment by ColeBantam** [comment]
Sadly I do not see any improvements in LanguageTool-20240315-snapshot.oxt. Black-Gui-issue still exists (again) and I had a crash right on first document loading (without stack trace).

What should be the advantage of the new toolbar? Because I see only disadvantages compared to the old style (until 3.3). Lots of GUI-load-problems and permanently resetting toolbar that has to be repositioned on every LT-Update :/ Up until 3.3, there never has been any issue with the toolbar...

FredKruse wrote: "I think it was a problem with a calc document, right?
On the mentioned crash, there was a calc and a writer document open. Not sure which one caused the crash, but I think the calc one was in foreground when it crashed.

**Comment by FredKruse** [comment]
I switched back to the static toolbar as in LT 6.3.1.
The issue should be solved. Please test it with tomorrow's snapshot and provide feedback.

**Comment by ColeBantam** [comment]
Thanks for rolling back to the old toolbar-style. All the GUI-issues are gone now :)

Still have an eye on stability though, as I again had a LibO-crash without stack-trace...

**Comment by FredKruse** [comment]
I changed the code a little to increase stability. Please test it with tomorrow's snapshot.

**Comment by ColeBantam** [comment]
ok. have LanguageTool 6.4-SNAPSHOT (2024-03-22 18:59:04 +0100, cfa9f5b) installed for testing now. No issues during the first short basic tests.
